src/seedu/addressbook/data/person/Address.java

```diff
@@ -38,14 +49,19 @@ public static boolean isValidAddress(String test) {
 
     @Override
     public String toString() {
-        return value;
+        return block + ", " + street + ", " + unit + ", " + postalCode;
```

Comment:
Good work here. Using the toString() for the respective parts is a cleaner implementation.